cocotb/utils.py

```diff
@@ -502,6 +502,8 @@ def want_color_output():
         want_color = True
     if os.getenv("GUI", default='0') == '1':
         want_color = False
+    if os.getenv("NO_COLOR") is not None:
+        want_color = False
```

Comment:
My reading of the link you reference is that `COCOTB_ANSI_OUTPUT` should take precedence over this - so this `if` should be first, not last.

Reply:
@eric-wieser Thank you for your answer. 

From he website:
> All command-line software which outputs text with ANSI color added should check for the presence of a NO_COLOR environment variable that, when present (regardless of its value), prevents the addition of ANSI color.

I read this (as non native English speaker) as: If NO_COLOR is set, there should be no case where color is still possible. At the moment, if NO_COLOR=1 and COCOTB_ANSI_OUTPUT=1 then there will be no color. 

From the original Issue #1309  @imphil 
> We should support this variable, and deprecate COCOTB_ANSI_OUTPUT (don't remove it just yet).

Can you explain your reading of the topic?
Thank you

Reply:
Maybe we should output a warning if the two contradicting variables are set

Reply:
Also from the website:

> How should configuration files and command-line arguments be processed in the presence of $NO_COLOR?
> 
> User-level configuration files and per-instance command-line arguments should override $NO_COLOR. A user should be able to export $NO_COLOR in their shell configuration file as a default, but configure a specific program in its configuration file to specifically enable color.
> 
> This also means that software that can add color but doesn’t by default does not need to care about $NO_COLOR, because it will only ever be adding color when instructed to do so (as it should be).

My interpretation is that `COCOTB_ANSI_OUTPUT` should be considered a "per-instance command-line argument", since we pass most of our options via env vars.

Reply:
Thank you. Will change this